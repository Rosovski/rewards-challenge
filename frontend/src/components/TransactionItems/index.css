table {
  width: 80%;
  border-collapse: collapse;
}

th {
  height: 40px;
  text-align: left;
}

table,
td,
th {
  border: 1px solid black;
}

th,
td {
  padding: 15px;
}
