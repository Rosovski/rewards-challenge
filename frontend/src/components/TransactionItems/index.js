import React from "react";
import useFetch from "../../utils/hooks/useFetch";
import "./index.css";

const TransactionItems = ({ customerId }) => {
  const transactionsData = useFetch(`/api/transactions/${customerId}`);

  return (
    <div data-testid="transaction-items-component">
      <table data-testid="transaction-table">
        <tbody>
          <tr>
            <th>Id</th>
            <th>Amount</th>
            <th>Bill Date</th>
          </tr>
          {transactionsData &&
            transactionsData.map((item, i) => (
              <tr key={i}>
                <td>{item.Id}</td>
                <td>{item.amount}</td>
                <td>{item.billDate}</td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

export default TransactionItems;
