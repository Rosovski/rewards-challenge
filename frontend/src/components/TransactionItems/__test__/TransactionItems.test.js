import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import TransactionItems from "..";
import apiClient from "../../../utils/apiClient";

afterEach(cleanup);

jest.mock("../../../utils/apiClient");

describe("test suite for the component of transaction table", () => {
  it("render the component", () => {
    render(<TransactionItems customerId="1" />);
  });

  it("mock the api call to get response", async () => {
    apiClient.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: {
          result: [
            { Id: "1", amount: 145, billDate: "2023-01-13" },
            { Id: "1", amount: 145, billDate: "2022-12-13" },
            { Id: "1", amount: 145, billDate: "2022-11-13" },
            { Id: "1", amount: 145, billDate: "2022-10-13" },
            { Id: "1", amount: 145, billDate: "2022-09-13" },
          ],
        },
      })
    );

    const { findByText, findByTestId } = render(
      <TransactionItems customerId="1" />
    );

    const table = await findByTestId("transaction-table");
    const id = await findByText(/Id/);
    const amount = await findByText(/Amount/);
    const billDate = await findByText(/Bill Date/);

    expect(table).toBeInTheDocument();
    expect(id).toBeInTheDocument();
    expect(amount).toBeInTheDocument();
    expect(billDate).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<TransactionItems customerId="1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
