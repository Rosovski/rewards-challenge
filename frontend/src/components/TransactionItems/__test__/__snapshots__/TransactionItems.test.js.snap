// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for the component of transaction table matches snapshot 1`] = `
<div
  data-testid="transaction-items-component"
>
  <table
    data-testid="transaction-table"
  >
    <tbody>
      <tr>
        <th>
          Id
        </th>
        <th>
          Amount
        </th>
        <th>
          Bill Date
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
